genericOidc.go

```diff
@@ -37,14 +41,20 @@ func NewGenericOIDC(ic *IssuerConfig, opts ...GenericOIDCOption) (*GenericOIDC,
 
 	cfg := &GenericOIDCCfg{
 		UserExtractorFn:      DefaultGenericUserExtractor,
+		Timeout:              10 * time.Second,
 		SupportedSigningAlgs: []string{"RS256", "RS384", "RS512"},
 	}
 
 	for _, opt := range opts {
 		opt(cfg)
 	}
 
+	client := &http.Client{
+		Timeout: cfg.Timeout,
+	}
 	ctx := context.Background()
+	ctx = context.WithValue(ctx, oauth2.HTTPClient, client)
+
```

Comment:
Normally `http.Client`s are for global usage. If you use then for *short shots* you should set the transports `DisableKeepAlives: true` otherwise you will leak memory because the transport will have its internal state management with a goroutine. 
I'm unsure here, because i don't know how often the `NewGenericOIDC` itself will be called and if the returned `GenericOIDC`s are global values by themself. But i don't think you will need a keepalive functionality here, so it should work without keepalive too and the internal state management of the transport is not needed here?

Reply:
Good point, it will only be used for key requests - disabled keepalives.